tools/esdlcmd/esdlcmd_monitor.cpp

```diff
@@ -1001,39 +914,44 @@ class EsdlDiffGenCmd : public EsdlConvertCmd
 
         respTemplate = monitoringTemplate->queryPropTree(resp_type);
 
-        xpath.setf("EsdlStruct[@name='%s']", resp_type.str());
-        IPropertyTree *respTree = depTree->queryPropTree(xpath);
-        if (respTree)
-        {
-            respTree = checkExtractNestedResponseType(depTree, respTree, resp_type);
+        IPropertyTree *respTree = depTree->queryPropTree(xpath.setf("EsdlResponse[@name='%s']", resp_type.str()));
```

Comment:
if the parameter is a char * it automatically casts.  If its a va_list (...) then you always have to use s.str().